### ISSUE TITLE
Pwsh.exe maxing out memory over time. 

### ISSUE THREAD

**Issue by Byrd1083** [MultiPoolMiner/MultiPoolMiner#2106]
I noticed today the rigs lagging down really bad. I looked at task manager to see over 250 pwsh.exe open and the ram use just continues to rise a bit per second steadily. Why is the program opening new instances without closing old ones cause the system to run out of memory the longer it runs? It forces me to restart or I can not remote login from to slow of system. When it is restarted and mining, MPM is only using like 500mb. One rig has been going for 24 and is now using almost all the ram and still rising. Any ideas?

FYI, I have already set page file size to 60gb~80gb.

**Comment by UselessGuru** [comment]
I cannot reproduce this issue.

What are the command lines that were executed with all these powershell instances? (Taks Manager -> Details -> Select columns -> tick 'Command line'

**Comment by grantemsley** [comment]
Do you have the log viewer window open?  That can eat a ton of memory and open hundreds of powershell processes.

**Comment by Byrd1083** [comment]
Just checked this moring from restart last night. within 12 hours I have 284 pwsh.exe instances open and about 3000mb being used. Here is my MPM zipped.



[ZIP.zip](https://github.com/MultiPoolMiner/MultiPoolMiner/files/2324051/ZIP.zip)
![taskm-screenshot](https://user-images.githubusercontent.com/42227809/44663312-07d74500-a9c5-11e8-95d2-ca5d8dfdbe73.jpg)

**Comment by Byrd1083** [comment]
What is the point of having a log window if it crashes ram? This needs to be fixed please.

**Comment by grantemsley** [comment]
Yes, the log viewer is broken. It will be replaced with something better.  For now, just remove the line for reader.ps1 in your start.bat file.

**Comment by UselessGuru** [comment]
@Byrd1083 
> What is the point of having a log window if it crashes ram

Before we start barking up the wrong tree:

What are the command lines that were executed with all these powershell instances? (Task Manager -> Details -> Select columns -> tick 'Command line'

**Comment by robl45** [comment]
I have same issue.  Just had 34 of them.  Removed the line with reader.ps1 in it as mentioned above and clearly it didn't make a difference.  I've had this issue for a long time, I thought it was a well known issue.  It started when the logging and everything was incorporated instead of just separate miner windows.

**Comment by UselessGuru** [comment]
It could as well be the hidden miner windows.
Does the error persists if you start MPM with -ShowMinerWindows?

**Comment by Byrd1083** [comment]
I've been playing around with rainbow miner. It appears to be based on similar code. It displays log windows properly with out this issue. Someone should pop over to rainbow miner and see if it can be propagated over.

**Comment by Byrd1083** [comment]
It is still in development mode but the app has some nice upgraded features.

**Comment by UselessGuru** [comment]
Please try the latest 3.0.1 build. There have been changes to the downloader which start hidden PWSH processes too.

**Comment by carlo0000** [comment]
hi
i update to the last version on 1 computer with 2Gb ram
but i can't finish the benchmark as powershell use all the memory and hard drive works 100% and get computer very slow to access
previous installed multipool version was 2.x

on my other's computer running the older version of 3.0.1 (2 or 4Gg of ram no issue so far for 2 days)

i always running with -ShowMinerWindow
i gonna test with removing ligne reader.ps1

**Comment by UselessGuru** [comment]
> it won't close the excavador miner

Version 3.1.0 will close Excavator when no longer needed

**Comment by UselessGuru** [comment]
This might help:

https://github.com/MultiPoolMiner/MultiPoolMiner/issues/2222
